web/api/user/utilities.py

```diff
@@ -1,10 +1,35 @@
+import os
 from PIL import Image
 from .artpiece import Artpiece
+import random
 
 #function to pull image off of database
 def pull_picture(id):
     # handle invalid id
     artpiece = Artpiece.get_by_id(id)
     image = Image.frombytes("RGBX", (616, 414), artpiece.raw_image)
     image.show()
-    return image
\ No newline at end of file
+    return image
+
+"""
+Return a list of images in the 'gallery' folder and their descriptions
+Output is list of tuples (image_location, image_description)
+    output list is in random order for random display order every time
+"""
+def get_gallery_images():
+    internal_gallery_path = './web/static/img/gallery/'
+    public_gallery_path = '/static/img/gallery/'
+
+    imagename_list = [internal_gallery_path + filename for filename in os.listdir(internal_gallery_path)]
+
+    desc_list = list()
+    for img_name in imagename_list:
+        with Image.open(img_name) as image:
+            desc_list.append(image.getexif().get(270)) #Exif ID 270 = ImageDescription
+            #now check aspect ratio and normalize
```

Comment:
Can we manually size the images with the same aspect ratio before uploading to git? Preferably the same dimensions too.

Reply:
The dimension thing killed me - two different cameras used two different aspect ratios and resolutions. Developing a recommended standard would help, but I came to the same conclusion I think you're expressing here: don't handle it in the code, and certainly not at runtime.

I'll make a new issue for developing/checking a standard image ratio/resolution, and remove this comment.